packages/jaeger-ui/src/utils/tracking/ga.tsx

```diff
@@ -24,6 +23,40 @@ import { logTrackingCalls } from './utils';
 import { getAppEnvironment, shouldDebugGoogleAnalytics } from '../constants';
 import parseQuery from '../parseQuery';
 
+// Modify the global scope to add the `gtag` function and the `dataLayer` array
+declare global {
```

Comment:
What effect does this have on other code? Will the other code start seeing `window` as having a `dataLayer` property? Why do we want that, shouldn't this knowledge be private to this module?

Reply:
I don't think this knowledge should remain private to this module, as the `dataLayer` attribute is globally attached to the `window` object. Keeping it private means that all the other modules can assume there is no property named `dataLayer` on `window`, and thus if needed, they can add that themselves, which will mess with our current implementation.

Reply:
On second thoughts, the most ideal implementation would be for `dataLayer` to have the appropriate types within `ga.ts` and have the type `never` outside it so that no other module tries to access or modify the same.